### ISSUE TITLE
LDAP3 not able to add Active directory member to a group

### ISSUE THREAD

**Issue by rijubhaskergslab** [cannatag/ldap3#711]
I am trying to use below POC to add memeber to a group But it is giving error {'result': 32, 'description': 'noSuchObject', 'dn': '', 'message': "00000525: NameErr: DSID-031A1292, problem 2001 (NO_OBJECT), data 0, best match of:\n\t''\n\x00", 'referrals': None, 't ype': 'modifyResponse'}
POC below ----

from ldap3 import Server, ALL, Connection, SUBTREE, MODIFY_REPLACE,MODIFY_ADD, BASE, DEREF_NEVER
from ldap3.extend.microsoft.addMembersToGroups import ad_add_members_to_groups as addUsersInGroups

def connect():
method_name = "connect()"
ldap_host = 'XXXXXXXXXX'
ldap_port = 'XXXXXX'
username = 'XXXXXXXXX'
password = 'xxxxxxxxxxx'
protocol = 'ldap'
connection_url = protocol + "://" + ldap_host + ":" + ldap_port
server = Server(connection_url, get_info=ALL)
conn = Connection(server, username, password, auto_bind=True)
return conn

def update_user(userdn:str,group_dn:str):
print(userdn)
print(group_dn)
conn = connect()
userdn_list = [userdn]
groupdn_list = [group_dn]
updatedItem = conn.extend.microsoft.add_members_to_groups(userdn, group_dn)
#updatedItem = addUsersInGroups(conn, userdn_list, groupdn_list,raise_error=True)
print(updatedItem)
return conn.result

dn_user = 'CN=DemoRjF1 DemoRjL1,OU=Demo,OU=Users,OU=mycompany,DC=mydept,DC=com'
dn_group = 'CN=Wifi,OU=Groups,OU=mycompany,DC=mydept,DC=com'

upd = update_user(dn_user,dn_group)
print(upd)

Output -
False
{'result': 32, 'description': 'noSuchObject', 'dn': '', 'message': "00000525: NameErr: DSID-031A1292, problem 2001 (NO_OBJECT), data 0, best match of:\n\t''\n\x00", 'referrals': None, 't ype': 'modifyResponse'}

**Comment by cannatag** [comment]
Hi, do you still have this problem?

**Comment by Trojan295** [comment]
I would like to reopen the issue. I have exactly the same problem.

I have a setup with two domains with two-way trust and I would like to add a user from domain A to a group in domain B. The trust is configured properly, cause I'm able to add users using Microsoft tools

```
conn = ldap3.Connection(ldap3.Server('...', allowed_referral_hosts=[('*', True)]), auto_bind=True, user='...', password=...', auto_referrals=True, raise_exceptions=True,  authentication=ldap3.NTLM)

conn.search('...', '(objectclass=person)')
user_dn = conn.entries[0].entry_dn

conn.search('...', '(objectclass=group)')
group_dn = conn.entries[0].entry_dn

conn.extend.microsoft.add_members_to_groups(user_dn, group_dn)
```

results in:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/damian/.local/lib/python3.7/site-packages/ldap3/extend/__init__.py", line 305, in add_members_to_groups
    fix=fix)
  File "/home/damian/.local/lib/python3.7/site-packages/ldap3/extend/microsoft/addMembersToGroups.py", line 81, in ad_add_members_to_groups
    result = connection.modify(group, changes)
  File "/home/damian/.local/lib/python3.7/site-packages/ldap3/core/connection.py", line 1071, in modify
    response = self.post_send_single_response(self.send('modifyRequest', request, controls))
  File "/home/damian/.local/lib/python3.7/site-packages/ldap3/strategy/sync.py", line 121, in post_send_single_response
    responses, result = self.get_response(message_id)
  File "/home/damian/.local/lib/python3.7/site-packages/ldap3/strategy/base.py", line 377, in get_response
    raise LDAPOperationResult(result=result['result'], description=result['description'], dn=result['dn'], message=result['message'], response_type=result['type'])
ldap3.core.exceptions.LDAPNoSuchObjectResult: LDAPNoSuchObjectResult - 32 - noSuchObject - None - 00000525: NameErr: DSID-031A1292, problem 2001 (NO_OBJECT), data 0, best match of:
	''
 - modifyResponse - None
```
```
>>> conn.result
{'result': 32, 'description': 'noSuchObject', 'dn': '', 'message': "00000525: NameErr: DSID-031A1292, problem 2001 (NO_OBJECT), data 0, best match of:\n\t''\n\x00", 'referrals': None, 'type': 'modifyResponse'}
```

**Comment by Trojan295** [comment]
OK, the problem was with the two domains - it's not possible to add a person from one forest to a local group in an another forest. I had to use ForeignSecurityPrincipals with objectSid's and now it's working fine.

**Comment by pard68** [comment]
I'm actually getting this same issue.

I can search and find the user. with `search`. I can search and find the group with `search`. I have verified that the user is not in the group. running `extend.microsoft.add_users_to_groups(user, group)` returns false and `conn.result` returns `{'result': 32, 'description': 'noSuchObject', 'dn': '', 'message': "00000525: NameErr: DSID-031A1292, problem 2001 (NO_OBJECT), data 0, best match of:\n\t''\n\x00", 'referrals': None, 'type': 'modifyResponse'}`

I've tried running this on each of our domain controllers, ie `server = ldap3.Server("dc01.forest.example.com")`, as well as (and I lack the right name for this) the FQDN that resolves to any of our servers, ie `server = ldap3.Server("forest.example.com")`. At a loss for what it's not seeing.
